src/swap/calculateEthForExactNxm.test.ts

```diff
@@ -0,0 +1,65 @@
+import { parseEther } from 'viem';
+import { Reserves } from './reserves.type';
+import { calculateEthForExactNxm } from './calculateEthForExactNxm';
+import { BigNumber } from 'ethers';
+
+describe('calculateEthForExactNxm', () => {
+  const reserves: Reserves = {
+    nxmA: 142858457219554100789497n,
+    nxmB: 328817222320643252285179n,
+    ethReserve: 5000000000000000000000n,
+    budget: 43835000000000000000000n,
+  };
+
+  const cases = [
+    ['unit value', '1', 34999922981378727n],
+    ['decimal value', '0.1', 3499970248373829n],
+    ['large value', '1000000', 35006677919768798199868n], // ! this returns a negative value - why?
```

Comment:
⚠️ Tests fail for large values like `1,000,000` - the method returns a negative value; why?